apl-api2/src/main/resources/yaml/apollo-api-v2.yaml

```diff
@@ -623,587 +626,1649 @@ paths:
+        400:
+          $ref: '#/components/responses/BadRequest'
+        500:
+          $ref: '#/components/responses/ServerError'
+
+  /@@smc/{address}/state:
+    get:
+      tags:
+        - smc
+      summary: Get the contract state json given address.
+      description: Returns the contract state json.
+      operationId: getSmcStateByAddress
+      parameters:
+        - name: address
+          in: path
+          description: The contract address
+          required: true
+          schema:
+            type: string
+            example: 'APL-X5JH-TJKJ-DVGC-5T2V8'
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ContractStateResponse'
+        400:
+          $ref: '#/components/responses/BadRequest'
+        500:
+          $ref: '#/components/responses/ServerError'
+
+  /@@smc/{address}/spec:
+    get:
+      tags:
+        - smc
+      summary: Get the contract specification.
+      description: Returns the contract specification.
+      operationId: getSmcSpecificationByAddress
+      parameters:
+        - name: address
+          in: path
+          description: The contract address
+          required: true
+          schema:
+            type: string
+            example: 'APL-X5JH-TJKJ-DVGC-5T2V8'
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ContractSpecResponse'
+        400:
+          $ref: '#/components/responses/BadRequest'
+        500:
+          $ref: '#/components/responses/ServerError'
+
+  /@@smc/asr:
+    get:
+      tags:
+        - smc
+        - asr
+      summary: Get the ASR templates list.
+      description: Returns the ASR templates list.
+      operationId: getSmcAsrModules
+      parameters:
+        - name: language
+          in: query
+          description: The language name
+          required: false
+          schema:
+            type: string
+            example: 'js'
+        - name: version
+          in: query
+          description: The language version
+          required: false
+          schema:
+            type: string
+            example: '0.1.1'
+        - name: type
+          in: query
+          description: The module type, allowed values - token, escrow; the default value is token.
+          required: false
+          schema:
+            type: string
+            example: 'token'
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ModuleListResponse'
+        400:
+          $ref: '#/components/responses/BadRequest'
+        500:
+          $ref: '#/components/responses/ServerError'
+
+  /@@smc/asr/{module}/init:
```

Comment:
Probably better {/smc/asr/{module}/creators}  to conform rest API best practices